macaw-loader/src/Data/Macaw/BinaryLoader.hs

```diff
@@ -6,74 +6,83 @@
-               , memoryEndianness :: MM.Endianness
-               , archBinaryData :: ArchBinaryData arch binFmt
-               , binaryFormatData :: BinaryFormatData arch binFmt
-               , loadDiagnostics :: [Diagnostic arch binFmt]
-               , binaryRepr :: BinaryRepr binFmt
-               , originalBinary :: binFmt
-               , loadOptions :: LC.LoadOptions
-               }
-
--- | A class for architecture and binary container independent binary loading
---
--- An instance is required for every arch/format pair, but the interface is more
--- accessible to callers than some alternatives.
---
--- n.b. The (MM.MemWidth (MR.ArchAddrWidth arch)) constraint enables
--- the LoadedBinary output type to functions by simply expressing
--- (BinaryLoader arch binFmt) as a constraint on those functions, but
--- many of the usage sites will still need to express a constraint
--- like:
---
--- > (BinaryAddrWidth binFmt ~ MC.ArchAddrWidth arch) =>
---
--- It would be nice to add that constraint directly to the class here,
--- but the macaw-loader family should only depend on macaw-base and
--- whatever loader (e.g. elf), and adding the above constraint
--- requires definitions from the macaw architecture specific modules
--- (e.g. macaw-ppc), creating a dependency cycle (because macaw-ppc
--- should be able to use macaw-loader-ppc). Ergo this constraint must
--- be expressed at the usage sites instead.
-
-class (MM.MemWidth (MR.ArchAddrWidth arch)) =>
-      BinaryLoader arch binFmt where
+data LoadedBinary arch binFmt
+  = LoadedBinary
+  { memoryImage :: MM.Memory (MM.ArchAddrWidth arch)
+  , memoryEndianness :: MM.Endianness
+  , archBinaryData :: ArchBinaryData arch binFmt
+  , binaryFormatData :: BinaryFormatData arch binFmt
+  , loadDiagnostics :: [Diagnostic arch binFmt]
+  , binaryRepr :: BinaryRepr binFmt
+  , originalBinary :: binFmt
+  , loadOptions :: LC.LoadOptions
+  }
+
+{- | A class for architecture and binary container independent binary loading
+
+An instance is required for every arch/format pair, but the interface is more
+accessible to callers than some alternatives.
+
+n.b. The (MM.MemWidth (MR.ArchAddrWidth arch)) constraint enables
+the LoadedBinary output type to functions by simply expressing
+(BinaryLoader arch binFmt) as a constraint on those functions, but
+many of the usage sites will still need to express a constraint
+like:
+
+> (BinaryAddrWidth binFmt ~ MC.ArchAddrWidth arch) =>
+
+It would be nice to add that constraint directly to the class here,
+but the macaw-loader family should only depend on macaw-base and
+whatever loader (e.g. elf), and adding the above constraint
+requires definitions from the macaw architecture specific modules
+(e.g. macaw-ppc), creating a dependency cycle (because macaw-ppc
+should be able to use macaw-loader-ppc). Ergo this constraint must
+be expressed at the usage sites instead.
+-}
+class
+  (MM.MemWidth (MR.ArchAddrWidth arch)) =>
+  BinaryLoader arch binFmt
+  where
```

Comment:
It's hard to tell if this was simply reformatting the declaration and changing the comment markings for the haddock or if there were actual changes applied.  It would be better to make purely format changes in a separate commit (and possibly a separate PR) which would allow easier review of substantive changes and also allow rollback without entanglements.